File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/swift/F15.P3.swift
// BEGIN PROBE F15.P3
protocol Named { func tag() -> String }
struct A: Named { func tag() -> String { "a" } }
struct B: Named { func tag() -> String { "b" } }
let items: [Named] = [A(), B()]
let firstTag = items[0].tag()
// END PROBE F15.P3

print(firstTag)

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F15.P3.rs
fn first_tag() -> &'static str {
// BEGIN PROBE F15.P3
trait Named { fn tag(&self) -> &str; }
struct A;
impl Named for A { fn tag(&self) -> &str { "a" } }
struct B;
impl Named for B { fn tag(&self) -> &str { "b" } }
let items: Vec<&dyn Named> = vec![&A, &B];
items[0].tag()
// END PROBE F15.P3
}

fn main() {
    println!("{}", first_tag());
}
